app/controllers/AddBusinessStartDateCheckController.scala

```diff
@@ -18,55 +18,155 @@ package controllers
+  lazy val backUrlAgent: String = routes.AddBusinessStartDateController.showAgent().url
+
+  lazy val addBusinessTradeUrl: String = routes.AddBusinessTradeController.show().url
+  lazy val addBusinessTradeAgentUrl: String = routes.AddBusinessTradeController.showAgent().url
+
+  lazy val homePageCall: Call = routes.HomeController.show()
+  lazy val homePageCallAgent: Call = routes.HomeController.showAgent
 
   def show(): Action[AnyContent] = (checkSessionTimeout andThen authenticate andThen retrieveNino
     andThen retrieveIncomeSources andThen retrieveBtaNavBar).async {
     implicit user =>
-      handleRequest(isAgent = false)
+      handleRequest(
+        isAgent = false,
+        backUrl = backUrl,
+        homePageCall = homePageCall,
+        itvcErrorHandler = itvcErrorHandler
+      )
+  }
+
+  def showAgent(): Action[AnyContent] = Authenticated.async {
+    implicit request =>
+      implicit user =>
+        getMtdItUserWithIncomeSources(incomeSourceDetailsService) flatMap {
+          implicit mtdItUser =>
+            handleRequest(
+              isAgent = true,
+              backUrl = backUrlAgent,
+              homePageCall = homePageCallAgent,
+              itvcErrorHandler = itvcErrorHandlerAgent
+            )
+        }
   }
 
-  def showAgent(): Action[AnyContent] =
-    Authenticated.async {
-      implicit request =>
-        implicit user =>
-          getMtdItUserWithIncomeSources(incomeSourceDetailsService) flatMap {
-            implicit mtdItUser =>
-              handleRequest(isAgent = true)
-          }
-    }
+  def submit(date: String): Action[AnyContent] = (checkSessionTimeout andThen authenticate andThen retrieveNino
+    andThen retrieveIncomeSources andThen retrieveBtaNavBar).async {
+    implicit request =>
+      handleSubmitRequest(
+        date = date,
+        isAgent = false,
+        backUrl = backUrl,
+        nextPageUrl = addBusinessTradeUrl,
+        itvcErrorHandler = itvcErrorHandler
+      )
+  }
 
-  //TODO Start date page to be implemented
-  def handleRequest(isAgent: Boolean)(implicit user: MtdItUser[_], ec: ExecutionContext): Future[Result] = {
-    Future {
+  def submitAgent(date: String): Action[AnyContent] = Authenticated.async {
+    implicit request =>
+      implicit user =>
+        getMtdItUserWithIncomeSources(incomeSourceDetailsService) flatMap {
+          implicit mtdItUser =>
+            handleSubmitRequest(
+              date = date,
+              isAgent = true,
+              backUrl = backUrlAgent,
+              nextPageUrl = addBusinessTradeAgentUrl,
+              itvcErrorHandler = itvcErrorHandlerAgent
+            )
+        }
+  }
+
+  def handleRequest(isAgent: Boolean,
+                    backUrl: String,
+                    homePageCall: Call,
+                    itvcErrorHandler: ShowInternalServerError)
+                   (implicit request: Request[_], mtdItUser: MtdItUser[_]): Future[Result] = {
+    Future.successful(
```

Comment:
Please wrap only Redirect part with Future.success;
On line 126 you should use actual Future instead